transaction/std/src/transaction_builder.rs

```diff
@@ -3151,6 +3202,10 @@ pub mod transaction_builder_tests {
             funding_transaction_builder.add_input(funding_input_credentials);
 
             // Fund gift code TxOut
+            // FIXME: This should be `.add_gift_code_output` or someting, so that
+            // it goes to the gift code subaddress, but the fog hint is using the
+            // default subaddress
+            // (or, make a special builder for gift code funding transactions?)
```

Comment:
@iamalwaysuncomfortable ^

Reply:
this is now https://github.com/mobilecoinfoundation/mobilecoin/issues/2003

Reply:
Where's the codepath that generates the FOG hint for the default subaddress rather than the default subaddress?

The address passed to the fog_hint_address field is also the gift code subaddress
```rust
        let result = self.add_output_with_fog_hint_address(
            amount,
            recipient,
            recipient,
```